pkg/router/template/util/haproxy/map_entry_test.go

```diff
@@ -811,7 +811,11 @@ func TestGenerateCertConfigMapEntry(t *testing.T) {
 		}
 
 		certHost := templateutil.GenCertificateHostName(host, wildcard)
-		return &HAProxyMapEntry{Key: key, Value: certHost}
+		e := &HAProxyMapEntry{Key: key, Value: certHost}
+		if hascert {
+			e.SSLBindConfig = "[alpn h2,http/1.1]"
```

Comment:
Should we deduce this from the certificate count in the template itself?

Reply:
This approach looks fine.  Putting the logic in the template would be extra hassle for little benefit.